query/src/storages/fuse/table_functions/fuse_segments/fuse_segment.rs

```diff
@@ -0,0 +1,105 @@
+//  Copyright 2021 Datafuse Labs.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+use std::sync::Arc;
+
+use common_datablocks::DataBlock;
+use common_datavalues::prelude::*;
+use common_exception::Result;
+
+use crate::sessions::QueryContext;
+use crate::storages::fuse::io::MetaReaders;
+use crate::storages::fuse::meta::TableSnapshot;
+use crate::storages::fuse::FuseTable;
+
+pub struct FuseSegment<'a> {
+    pub ctx: Arc<QueryContext>,
+    pub table: &'a FuseTable,
+    pub snapshot_id: String,
+}
+
+impl<'a> FuseSegment<'a> {
+    pub fn new(ctx: Arc<QueryContext>, table: &'a FuseTable, snapshot_id: String) -> Self {
+        Self {
+            ctx,
+            table,
+            snapshot_id,
+        }
+    }
+
+    pub async fn get_history(&self) -> Result<DataBlock> {
```

Comment:
Shall we rename this method to `[try_]get_segment_info_as_data_block`(may be too long), since segment has no  history

Reply:
Just noticed that, will change it, thanks.